tests/js/common/shell/shell-collection-computed-values.js

```diff
@@ -72,6 +72,50 @@ function ComputedValuesAfterCreateCollectionTestSuite() {
       // computed value must be null
       assertNull(collection.toArray()[0].newValue);
     },
+
+    testKeepNullTrue2: function() {
+      collection.properties({
+        computedValues:
+          [{
+            name: "newValue",
+            expression: "RETURN IS_STRING(@doc.name.first) AND IS_STRING(@doc.name.last) ? " +
+              "MERGE(@doc.name, { full: CONCAT_SEPARATOR(' ', @doc.name.first, @doc.name.last) }) : null",
+            override: true,
+            keepNull: true
+          }]
+      });
+      if (isCluster) {
+        // unfortunately there is no way to test when the new properties
+        // have been applied on the DB servers. all we can do is sleep
+        // and hope the delay is long enough
+        internal.sleep(5);
+      }
+
+      const colProperties = collection.properties();
+      assertTrue(colProperties.hasOwnProperty("computedValues"));
+      assertEqual(colProperties.computedValues.length, 1);
+      assertTrue(colProperties.computedValues[0].hasOwnProperty("name"));
+      assertEqual(colProperties.computedValues[0].name, "newValue");
+      assertTrue(colProperties.computedValues[0].keepNull);
+
+      for (let i = 0; i < 50; ++i) {
+        collection.insert({name: {first: `a${i}`, last: `b${i}`}});
+      }
+      for (let i = 0; i < 50; ++i) {
+        collection.insert({name: {first: i, last: i}});
+      }
+      let amountNullValues = 0;
+      const docs = collection.toArray();
+      docs.forEach((doc, index) => {
+        assertTrue(doc.hasOwnProperty("newValue"));
+        if (index >= 50) {
```

Comment:
This test is not deterministic, as `toArray()` does not guarantee the order of the results.